enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -425,17 +438,14 @@ private URI startHaCommunication( LifeSupport haCommunicationLife, NeoStoreDataS
         return slaveHaURI;
     }
 
-    private boolean catchUpWithMaster()
+    private boolean catchUpWithMaster(UpdatePuller updatePuller)
             throws IllegalArgumentException, InterruptedException
     {
         monitor.catchupStarted();
         RequestContext catchUpRequestContext = requestContextFactory.newRequestContext();
         console.log( "Catching up with master. I'm at " + catchUpRequestContext );
 
-        // Unpause the update puller, because we know that we are a slave that just started communication with master.
-        UpdatePuller updatePuller = resolver.resolveDependency( UpdatePuller.class );
-        updatePuller.unpause();
-        if ( !updatePuller.await( UpdatePuller.NEXT_TICKET, false ) )
+        if ( !updatePuller.tryPullUpdates() )
```

Comment:
Previously `updatePuller.await()` could return false only when update puller was paused from listener. With this change update puller lives in `haCommunicationLife` (https://github.com/neo4j/neo4j/pull/5315/files#diff-fd737c728b4766f4bed4257af642e087R420) and `updatePuller.tryPullUpdates()` can return false only when `haCommunicationLife` is stopped. This happens when new slave switch is scheduled so it is ok to silently exit from the current slave switch.